ClothesApp/WebAPI/Controllers/BrandsController.cs

```diff
@@ -1,57 +1,101 @@
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<BrandDto>))]
+    public async Task<ActionResult<IList<BrandDto>>> GetAllBrands()
     {
-        var brandsDtos = await _brandService.GetAllBrands();
+        var brandDtos = await _brandService.GetAll();
 
-        return Ok(brandsDtos);
+        return Ok(brandDtos);
+    }
+    
+    [HttpGet("{id}/products")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<ProductDto>))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult<IList<ProductDto>>> GetBrandProducts([FromRoute] long id)
+    {
+        var productDtos = await _productService.GetProductsByBrandId(id);
+
+        return Ok(productDtos);
     }
     
     [HttpGet("{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BrandDto))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
     public async Task<ActionResult<BrandDto>> GetBrandById([FromRoute] long id)
     {
-        var brandDto = await _brandService.GetBrandById(id);
+        var brandDto = await _brandService.GetById(id);
         
         return Ok(brandDto);
     }
 
     [HttpDelete("{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BrandDto))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ProblemDetails))]
+    [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
     public async Task<IActionResult> DeleteBrandById([FromRoute] long id)
     {
-        var brandDto = await _brandService.DeleteBrandById(id);
+        var brandDto = await _brandService.DeleteById(id);
 
         return Ok(brandDto);
     }
 
     [HttpPost]
-    public async Task<ActionResult> AddBrand([FromBody] BrandNameDto brandNameDto)
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BrandDto))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult> AddBrand([FromBody] BrandInputDto brandInputDto)
     {
-        var brandDto = await _brandService.AddBrand(brandNameDto);
+        var brandDto = await _brandService.Add(brandInputDto);
         
         return Ok(brandDto);
     }
     
     [HttpPut("{id}")]
-    public async Task<ActionResult> UpdateBrand([FromRoute] long id, [FromBody] BrandNameDto brandNameDto)
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(BrandDto))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ProblemDetails))]
+    public async Task<ActionResult> UpdateBrand([FromRoute] long id, [FromBody] BrandInputDto brandInputDto)
     {
-        var brandDto = await _brandService.UpdateBrand(id, brandNameDto);
+        var brandDto = await _brandService.Update(id, brandInputDto);
         
         return Ok(brandDto);
     }
+    
+    [HttpPost("{brandId}/assign-product/{productId}")]
```

Comment:
Can't you just use UpdateProduct endpoint and provide brand it to it if you want to reassign brand?

Reply:
UpdateProduct endpoint makes more sense